eventlet/__init__.py

```diff
@@ -2,12 +2,12 @@
 import sys
 import warnings
 
-if sys.version_info < (3, 5):
+if sys.version_info < (3, 7):
     warnings.warn(
-        "Support for your Python version is deprecated and will be removed in the future",
+        "Your Python version is no longer supported by eventlet",
+        "Please consider upgrading to minimal supported Python version",
```

Comment:
this raises an error for me, you'll have to make the warning a single str arg

Reply:
Fixed.

```
>>> import eventlet
>>> eventlet.__version__
'0.34.2.dev1+gc2236b8.d20231222'
```

Reply:
Let me know if you still face the error.

Reply:
I must admit that originally I didn't actually test with python3.6, but just ran the warn() statement in py3.10, because I wanted to see how the multiline message would look like. With your original code, the issue would only be triggered when running with py3.6 or older. Now actually testing with py3.6, the error is gone, but also the deprecation warning actually isn't actually shown by default, since https://peps.python.org/pep-0565/ was only implemented for py3.7. So one would have to set something like `PYTHONWARNINGS=error` for the warning to actually take effect. This may just be the expected behavior and maybe I'm worrying too much about historic python versions, but maybe the alternative option of changing the warning to be a real error immediately might be a better solution?

Reply:
Thanks for your verification, and for your advices, they are really relevant.

Let's remove, from this pull request, the changes related to this warning. We can handle it in a separated pull request to more isolate the topics. I prefer to keep this pull request more focused on the sole topic of python version. I should have treated them separately.

Reply:
And anyway, this warning will be removed by https://github.com/eventlet/eventlet/pull/853/files
So, I suggest we do not waste time on it anywhere else.